### PR TITLE
1.21 Update port numbers

### PR DESCRIPTION
**Pull request by rnicoll**

### CODE REVIEW

File: src/chainparamsbase.cpp [hunk 1]
```diff
@@ -38,19 +38,19 @@ const CBaseChainParams& BaseParams()
...
 /**
- * Port numbers for incoming Tor connections (8334, 18334, 38334, 18445) have
+ * Port numbers for incoming Tor connections (22557, 44557, 55557, 33557) have
  * been chosen arbitrarily to keep ranges of used ports tight.
  */
 std::unique_ptr<CBaseChainParams> CreateBaseChainParams(const std::string& chain)
 {
     if (chain == CBaseChainParams::MAIN) {
-        return MakeUnique<CBaseChainParams>("", 8332, 8334);
+        return MakeUnique<CBaseChainParams>("", 22555, 22557);
```
Comment (DiegoRossiMafioletti) [inline src/chainparamsbase.cpp:47]: why are the doors hardcoded? (maybe using constants or macros would be better)

Comment (rnicoll) [reply src/chainparamsbase.cpp:47]: It's an odd choice in Bitcoin Core, but typically we don't refactor unnecessarily, to keep it easier to apply patches from upstream.

File: src/chainparams.cpp [hunk 2]
```diff
@@ -419,7 +419,7 @@ class CRegTestParams : public CChainParams {
         pchMessageStart[1] = 0xbf;
         pchMessageStart[2] = 0xb5;
         pchMessageStart[3] = 0xda;
-        nDefaultPort = 18444;
+        nDefaultPort = 33556;
```
Comment (langerhans) [inline src/chainparams.cpp:422]: Wouldn't it be annoying to have that different in 1.14 if you wanted to test interactions between them?

Comment (rnicoll) [reply src/chainparams.cpp]: Possibly. Mostly I don't care. Will revert :D

File: src/chainparamsbase.cpp [hunk 3]
```diff
@@ -38,19 +38,19 @@ const CBaseChainParams& BaseParams()
...
     } else if (chain == CBaseChainParams::TESTNET) {
-        return MakeUnique<CBaseChainParams>("testnet3", 18332, 18334);
+        return MakeUnique<CBaseChainParams>("testnet3", 44555, 44557);
     } else if (chain == CBaseChainParams::SIGNET) {
-        return MakeUnique<CBaseChainParams>("signet", 38332, 38334);
+        return MakeUnique<CBaseChainParams>("signet", 55555, 55557);
     } else if (chain == CBaseChainParams::REGTEST) {
-        return MakeUnique<CBaseChainParams>("regtest", 18443, 18445);
+        return MakeUnique<CBaseChainParams>("regtest", 33555, 33557);
```
Comment (patricklodder) [inline src/chainparamsbase.cpp:53]: Shouldn't this be around `18444` then per https://github.com/dogecoin/dogecoin/blob/e34197907f32f3fca371bfc490ef6e521723a070/src/chainparams.cpp#L440 ?

File: src/chainparamsbase.cpp [hunk 4]
```diff
@@ -38,19 +38,19 @@ const CBaseChainParams& BaseParams()
 }
 
 /**
- * Port numbers for incoming Tor connections (8334, 18334, 38334, 18445) have
+ * Port numbers for incoming Tor connections (22557, 44557, 55557, 33557) have
```
Comment (patricklodder) [inline src/chainparamsbase.cpp:41]: Comment needs to be updated to reflect port 18444?

**Review by patricklodder — CHANGES_REQUESTED** [review]
2 minor issues re: regtest ports, the rest looks good!

**Review by patricklodder — APPROVED** [review]
utACK!

**Pull request merged**